kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,151 @@
+      | select (.serverName == \"${serverName}\") | .serverStartPolicy)"
+    currentServerStartPolicy=$(echo ${domainJson} | jq "${extractPolicyCmd}")
+  fi
+  if [ -z "${currentServerStartPolicy}" ]; then
+    # Server start policy doesn't exist, add a new policy
+    addPolicyCmd=".[.| length] |= . + {\"serverName\":\"${serverName}\", \
+      \"serverStartPolicy\":\"${policy}\"}"
+    serverStartPolicyPatch=$(echo ${domainJson} | jq .spec.managedServers | jq -c "${addPolicyCmd}")
+  else
+    # Server start policy exists, replace policy value 
+    replacePolicyCmd="(.spec.managedServers[] \
+      | select (.serverName == \"${serverName}\") | .serverStartPolicy) |= \"${policy}\""
+    servers="(.spec.managedServers)"
+    serverStartPolicyPatch=$(echo ${domainJson} | jq "${replacePolicyCmd}" | jq -cr "${servers}")
+  fi
+  eval $__result="'${serverStartPolicyPatch}'"
+  eval $__currentPolicy=${currentServerStartPolicy}
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operatation string indicating whether to increment or decrement count
+# $4 - Return value containing replica update patch string
+# $5 - Retrun value containing updated replica count
+#
+function createReplicaPatch {
+domainJson=$1
+clusterName=$2
+operation=$3
+__result=$4
+__relicaCount=$5
+local errorMessage="@@ ERROR: Maximum number of servers allowed (maxReplica = ${maxReplicas}) \
+are already running. Please increase cluster size to start new servers."
+
+  replicasCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).replicas"
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$(($(echo ${domainJson} | jq "${replicasCmd}")-1))
+  elif [ "${operation}" == "INCREMENT" ]; then
+    replica=$(($(echo ${domainJson} | jq "${replicasCmd}")+1))
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -gt ${maxReplicas} ]; then
+      echo "${errorMessage}"
+      eval $__result="MAX_REPLICA_COUNT_EXCEEDED"
+      return
+    fi
+  fi
+
+  cmd="(.spec.clusters[] | select (.clusterName == \"${clusterName}\") \
+    | .replicas) |= ${replica}"
+  replicaPatch=$(echo ${domainJson} | jq "${cmd}" | jq -cr '(.spec.clusters)')
+  eval $__result="'${replicaPatch}'"
+  eval $__relicaCount="'${replica}'"
+}
+
+#
+# Function to validate whether a server belongs to a  cluster or is an independent managed server
+# $1 - Domain unique id.
+# $2 - Domain namespace.
+# $3 - Return value indicating if server is valid (i.e. if it's part of a cluster or independent server).
+# $4 - Retrun value containting cluster name to which this server belongs.
+#
+function validateServerAndFindCluster {
+domainUid=$1
+domainNamespace=$2 
+__isValidServer=$3
+__clusterName=$4
```

Comment:
It looks like vars above and below can all be declared local (ditto for other functions).

Reply:
Fixed.

Reply:
LGTM.